RATapi/classlist.py

```diff
@@ -333,11 +333,22 @@ def _check_unique_name_fields(self, input_list: Iterable[object]) -> None:
             Raised if the input list defines more than one object with the same value of name_field.
 
         """
-        names = [getattr(model, self.name_field).lower() for model in input_list if hasattr(model, self.name_field)]
-        if len(set(names)) != len(names):
-            raise ValueError(f"Input list contains objects with the same value of the {self.name_field} attribute")
+        error_list = []
+        new_names = [getattr(model, self.name_field) for model in input_list if hasattr(model, self.name_field)]
+        full_names = [name.lower() for name in new_names]
+        with contextlib.suppress(AttributeError):
+            full_names += [name.lower() for name in self.get_names()]
+        for i, name in enumerate(new_names):
+            if full_names.count(name.lower()) > 1:
+                error_list.append(f"    index {i} has the {self.name_field} '{name}', which is already specified")
```

Comment:
I was thinking about this, and the problem is that the list of indices you're referring to is not the indices resulting from the operation - if you're adding a list the indices refer to the list after adding, rather than before. Worse though, if you're inserting into the list, then the indices will be completely out of sync.

I am inclined to include indices though, so I'll see what I can come up with.

Reply:
Good shout on looking through unique names - I think the best way to do that is to loop through `list(set(new_names)`, then there's no need to check for this within the loop.